### PR TITLE
[CFE-417]: Feature - module detail

### PR DESCRIPTION
**Pull request by Poafs1**

### CODE REVIEW

File: src/lib/services/move/module/lcd.ts [hunk 1]
```diff
@@ -6,26 +6,26 @@ import type { Addr, IndexedModule, Nullable } from "lib/types";
...
-  baseEndpoint: string,
-  address: Addr,
+  endpoint: string,
+  vmAddress: Addr,
   moduleName: string
 ): Promise<IndexedModule> =>
   axios
     .get(
-      `${baseEndpoint}/initia/move/v1/accounts/${encodeURI(address)}/modules/${encodeURI(moduleName)}`
+      `${endpoint}/initia/move/v1/accounts/${encodeURIComponent(vmAddress)}/modules/${encodeURIComponent(moduleName)}`
```
Comment (evilpeach) [inline src/lib/services/move/module/lcd.ts:15]: `encodeURIComponent` to `encodeURI`

File: src/lib/services/move/module/index.ts [hunk 2]
```diff
@@ -274,13 +272,15 @@ export const useModuleData = (vmAddress: HexAddr, moduleName: string) => {
...
+      enabled,
     }
   );
 };
 
 export const useModuleTableCounts = (
   vmAddress: HexAddr,
-  moduleName: string
+  moduleName: string,
+  options: UseQueryOptions<ModuleTableCountsResponse> = {}
```
Comment (evilpeach) [inline src/lib/services/move/module/index.ts:283]: Why don't just pass enabled as before?

File: src/lib/services/move/module/index.ts [hunk 3]
```diff
@@ -257,7 +251,11 @@ export const useModules = (
   );
 };
 
-export const useModuleData = (vmAddress: HexAddr, moduleName: string) => {
+export const useModuleData = (
+  vmAddress: HexAddr,
+  moduleName: string,
+  enabled = true
+): UseQueryResult => {
```
Comment (evilpeach) [inline src/lib/services/move/module/index.ts:258]: You can remove `UseQueryResult`

File: src/lib/pages/module-details/index.tsx [hunk 4]
```diff
@@ -38,16 +49,27 @@ const ModuleDetailsBody = ({
...
+  const isFullTier = useTierConfig() === "full";
+  const fullData = useModuleData(vmAddress, moduleName, isFullTier);
+  const liteData = useModuleByAddressLcd({
+    address: vmAddress,
+    moduleName,
+    options: { enabled: !isFullTier },
+  });
+  const { data, isLoading } = (
+    isFullTier ? fullData : liteData
+  ) as UseQueryResult<ModuleData>;
```
Comment (evilpeach) [inline src/lib/pages/module-details/index.tsx:67]: Please don't make type assertion.

File: src/lib/services/move/module/lcd.ts [hunk 5]
```diff
@@ -6,26 +6,26 @@ import type { Addr, IndexedModule, Nullable } from "lib/types";
 import { parseWithError } from "lib/utils";
 
 export const getModuleByAddressLcd = async (
-  baseEndpoint: string,
-  address: Addr,
+  endpoint: string,
+  vmAddress: Addr,
```
Comment (songwongtp) [inline src/lib/services/move/module/lcd.ts:10]: If the endpoint supports both hex and bech, then keep it as address.

If not, change the type Addr to HexAddr

File: src/lib/services/move/module/lcd.ts [hunk 6]
```diff
@@ -6,26 +6,26 @@ import type { Addr, IndexedModule, Nullable } from "lib/types";
...
     .get(
-      `${baseEndpoint}/initia/move/v1/accounts/${encodeURI(address)}/modules/${encodeURI(moduleName)}`
+      `${endpoint}/initia/move/v1/accounts/${encodeURIComponent(vmAddress)}/modules/${encodeURIComponent(moduleName)}`
     )
     .then(({ data }) => parseWithError(zModuleResponseLcd, data).module);
 
 export const getModulesByAddressLcd = async (
   endpoint: string,
-  address: Addr
+  vmAddress: Addr
```
Comment (songwongtp) [inline src/lib/services/move/module/lcd.ts:21]: ditto

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Review by songwongtp — CHANGES_REQUESTED** [review]
let's flatten tables dir
- tables/
    - history/
    - ModuleRelatedProposalsTable.tsx
    - ModuleTxsTable.tsx
    - index.tsx (containing table-section)

File: src/lib/pages/module-details/index.tsx [hunk 7]
```diff
@@ -38,16 +49,27 @@ const ModuleDetailsBody = ({
...
   tab,
-}: ModuleDetailsQueryParams) => {
+}: ModuleDetailsQueryParams | ModuleDetailsLiteQueryParams) => {
   const router = useRouter();
   const navigate = useInternalNavigate();
   const formatAddresses = useFormatAddresses();
   const { hex: vmAddress } = formatAddresses(address);
 
-  const { data, isLoading } = useModuleData(vmAddress, moduleName);
+  const isFullTier = useTierConfig() === "full";
```
Comment (songwongtp) [inline src/lib/pages/module-details/index.tsx:58]: what about manually handle lite tabIndex here, so we don't have to create `TabIndexLite` and `zModuleDetailsLiteQueryParams`?

e.g.
```
const currentTab = !isFullTier && tab === TabIndex.TxsHistories ? TabIndex.Overview : tab;
```

up to debate @Poafs1 @evilpeach

**Review by evilpeach — APPROVED** [review]

File: src/lib/pages/module-details/components/ModuleInfoBody.tsx [hunk 8]
```diff
@@ -1,35 +1,43 @@
...
+          type="tx_hash"
+          value={recentPublishTransaction}
+          showCopyOnHover
+        />
       </LabelText>
     );
   }
 
-  if (proposal && proposal.id) {
+  if (recentPublishProposal && recentPublishProposal.id) {
```
Comment (songwongtp) [inline src/lib/pages/module-details/components/ModuleInfoBody.tsx:32]: `if (recentPublishProposal)` is enough, `id` can only be number

**Review by songwongtp — APPROVED** [review]

**Pull request merged**